src/Npgsql.NodaTime/TimestampHandler.cs

```diff
@@ -167,5 +173,14 @@ internal static void WriteDouble(Instant instant, NpgsqlWriteBuffer buf)
         }
 
         #endregion Write
+
+        DateTime INpgsqlSimpleTypeHandler<DateTime>.Read(NpgsqlReadBuffer buf, int len, FieldDescription fieldDescription)
+           => ((INpgsqlSimpleTypeHandler<DateTime>)_bclHandler).Read(buf, len, fieldDescription);
+
+        int INpgsqlSimpleTypeHandler<DateTime>.ValidateAndGetLength(DateTime value, NpgsqlParameter parameter)
+            => ((INpgsqlSimpleTypeHandler<DateTime>)_bclHandler).ValidateAndGetLength(value, parameter);
+
+        void INpgsqlSimpleTypeHandler<DateTime>.Write(DateTime value, NpgsqlWriteBuffer buf, NpgsqlParameter parameter)
+            => ((INpgsqlSimpleTypeHandler<DateTime>)_bclHandler).Write(value, buf, parameter);
     }
```

Comment:
You might be wondering why I have an explicit cast here.  This is because there is an implicit conversion from `DateTime` to `NpgsqlDateTime`. Therforce c#s overload resolution always picks `NpgsqlDateTime` methods in the BCL handler. Since there is special handling for DateTime.MaxValue/MinValue (convertInfinityDateTime), this matters.

Reply:
Thanks for the explanation, another problematic aspect of the provider-specific types... We should consider removing the implicit cast operators (unless we remove the types entirely...)